df_engine_extensions/src/dist_sql_query/resolver.rs

```diff
@@ -115,9 +117,14 @@ impl Resolver {
     }
 
     pub fn resolve_partitioned_scan_internal(
-        &self,
+        &mut self,
         plan: Arc<dyn ExecutionPlan>,
     ) -> DfResult<Arc<dyn ExecutionPlan>> {
+        // Check if this plan is `AnalyzeExec`, if it is, we should collect metrics.
+        if plan.as_any().downcast_ref::<AnalyzeExec>().is_some() {
+            self.is_analyze = true;
+        }
```

Comment:
`AnalyzeExec` is ensured to be the top-level `node`, we can check it in `resolve_partitioned_scan`, and make it a param of `resolve_partitioned_scan_internal`.